File: C++/src/two_sum.cpp
#include <two_sum.hpp>
#include <unordered_map>

std::vector<int> Solution::twoSum(std::vector<int> &nums, int target) {
  std::unordered_map<int, int> hashMap;
  for (int i = 0; i < (int)nums.size(); i++) {
    int remainder = target - nums[i];
    if (hashMap.find(remainder) != hashMap.end()) {
      return std::vector<int>{hashMap[remainder], i};
    }

    hashMap[nums[i]] = i;
  }
  return std::vector<int>{};
}

File: Go/two_sum/two_sum.go
package two_sum

func twoSum(nums []int, target int) []int {
	hashMap := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		remainder := target - nums[i]
		if value, ok := hashMap[remainder]; !ok {
			hashMap[nums[i]] = i
		} else {
			return []int{value, i}
		}
	}
	return []int{}
}
